pkg/systemd/terminal.jsx

```diff
@@ -81,8 +95,35 @@ const _ = cockpit.gettext;
         }
 
         async componentDidMount() {
+            cockpit.addEventListener("locationchanged", this.onNavigate);
             const user = await cockpit.user();
-            this.setState({ user, channel: this.createChannel(user) });
+            var dir
+            if (cockpit.location.options.path) {
+                const exists = await this.dirExists();
+                if (exists){
+                        dir = cockpit.location.options.path;
+                } else{
+                    this.setState({ pathError: cockpit.location.options.path })
+                }
+            }
+            this.setState({ user, channel: this.createChannel(user, dir)});
+        }
+
+        async dirExists(){
+            var exists
+            const cmmd = "test -d " + cockpit.location.options.path
+            await cockpit.script(cmmd, [], {err: "message"})
+                    .then(() => {
+                        exists = true
+                    })
+                    .catch((err) => {
+                       exists = false
+                    })
+            return exists
+        }
```

Comment:
This uses a shell command, so is subject to shell injection and all sorts of breakage if the (user-controlled) path includes spaces, `;` or `rm -rf ~` :fearful: Just imagine sending someone a crafted `http://localhost:9090` link.

Please use `fsinfo()` from pkg/lib/cockpit/fsinfo.ts for that. It avoids a shell and thus the security problems, and is much cheaper, too.